rdbms/src/main/java/org/spine3/server/storage/jdbc/util/IdHelper.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.jdbc.util;
+
+import org.spine3.Internal;
+import org.spine3.server.Entity;
+import org.spine3.server.Identifiers;
+import org.spine3.server.reflect.Classes;
+import org.spine3.server.storage.jdbc.DatabaseException;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Helps to work with entity IDs.
+ *
+ * @author Alexander Litus
+ */
+@Internal
+public abstract class IdHelper<I> {
+
+    private static final String TYPE_VARCHAR = "VARCHAR(999)";
+
+    private static final String TYPE_BIGINT = "BIGINT";
+
+    private static final String TYPE_INT = "INT";
+
+    private static final int ENTITY_OR_AGGREGATE_ID_TYPE_GENERIC_PARAM_INDEX = 0;
```

Comment:
This should not be part of this class. If we decide change this in `Entity` or `Aggregate`, it would break this class. 

Having this knowledge here makes the system fragile. Please watch the video of Uncle Bob on this.